MultiplayerTestRunner/src/Main.cpp

```diff
@@ -0,0 +1,115 @@
+#include "../include/ErrorCodes.h"
+#include "../include/ProcessDescriptors.h"
+#include "CLIArgs.h"
+#include "LoginRAII.h"
+#include "RunnableTests/CreateAvatar.h"
+#include "SpaceRAII.h"
+#include "Utils.h"
+
+#include <CSP/Common/String.h>
+#include <chrono>
+#include <string>
+#include <thread>
+
+#ifdef RUN_MULTIPLAYER_RUNNER_TESTS
+	#include <gtest/gtest.h>
+#endif
+
+/********************************
+ * MULTIPLAYER TEST RUNNER
+ * This is a CLI application designed to launch specified tests on independent processes (as CSP currently supports only one user per process), in
+ * order to support testing multiple clients on multiple processes interacting in the same room.
+ * When used in an automated testing context, it is
+ * designed to communicate back with the calling process via stdout process descriptors (see `include/ProcessDescriptors.h`), such that the calling
+ * process can reliably call assertions with the correct timings.
+ * The CLI itself provides some flexibility for use in a more direct manner as an ad-hoc test application.
+ *
+ * Dependencies :
+ *	- CSP. From the ConnectedSpacePlatform Solution. Linked.
+ *  - UUID_v4. For generating unique room names. Header only.
+ *  - CLI11. For defining the command line interface. Header only.
+ ********************************/
+
+/*
+ * Calls the appropriate test based on the provided test identifier stored in settings
+ * Will print the READY_FOR_ASSERTIONS_DESCRIPTOR when after the test code has been executed
+ * By this point, the client should be logged in and inside a space, hence tests need not concern themselves with space creation and cleanup.
+ * This function blocks until the timeout has elapsed.
+ */
+void RunTest(CLIArgs::RunnerSettings Settings, std::chrono::steady_clock::time_point ProgramStartTime)
+{
+	using namespace MultiplayerTestRunner::TestIdentifiers;
+	switch (Settings.TestIdentifier)
+	{
+		case TestIdentifier::CREATE_AVATAR:
+			CreateAvatar::RunTest();
```

Comment:
[question] is this requirement to manually add new tests in the switch maintainable long term? 
"I added a test but it won't run!!!! - Have you added it in the switch? - Doh!"

("yes" is an acceptable answer)

Reply:
"Yes"

:P

Seriously though I don't think it's worth engineering a solution to this sort of thing if and until the number of tests become unmanageable. If (big if :P) we've our warnings configured correctly, the build will error about non-exhaustive switch statements.

Reply:
that is completely fair enough